paho/client.go

```diff
@@ -641,133 +655,140 @@ func (c *Client) Subscribe(ctx context.Context, s *Subscribe) (*Suback, error) {
 		}
 	}
 
-	c.debug.Printf("subscribing to %+v", s.Subscriptions)
-
-	subCtx, cf := context.WithTimeout(ctx, c.PacketTimeout)
-	defer cf()
-	cpCtx := &CPContext{subCtx, make(chan packets.ControlPacket, 1)}
+	respInternal := make(chan packets.ControlPacket)
+	resp := make(chan packets.ControlPacket, 1)
 
 	sp := s.Packet()
 
-	mid, err := c.MIDs.Request(cpCtx)
+	var err error
+	sp.PacketID, err = c.MIDs.Request(&CPContext{context.TODO(), respInternal})
```

Comment:
I think that contexts should be removed from the packet ID tracker, but I put `context.TODO()` as a placeholder for now.